tracer/test/Datadog.Trace.ClrProfiler.IntegrationTests/TracingIntegrationTest.cs

```diff
@@ -38,5 +45,28 @@ public void ValidateIntegrationSpans(IEnumerable<MockSpan> spans, string metadat
                 }
             }
         }
+
+        internal async Task RunTestAndAssertTelemetry(Func<Task<MockTelemetryAgent>> actualTest, IntegrationId integrationToAssert)
+        {
+            // The server implementation of named pipes is flaky so have 3 attempts
+            var attemptsRemaining = 3;
+            while (true)
+            {
+                attemptsRemaining--;
+                var telemetry = await actualTest();
+                try
+                {
+                    // We know what the issue is - there's a shutdown bug, which fails during the final flush
+                    // We've grabbed memory dumps from it and we don't know how to fix it.
```

Comment:
Should probably double check I'm correct about this - I frequently get mixed up with the various runtime bugs 😅